packages/create-cloudflare/src/templateMap.ts

```diff
@@ -1,53 +1,368 @@
+	) => Promise<Record<string, string | object>>;
+
+	path?: string;
+
+	testFlags?: string[];
+	compatibilityFlags?: string[];
+	deployCommand?: string[];
+	devCommand?: string[];
 };
 
-export const templateMap: Record<string, TemplateConfig> = {
-	"hello-world": {
-		label: `"Hello World" Worker`,
-		handler: runWorkersGenerator,
-	},
-	webFramework: {
-		label: "Website or web app",
-		handler: runPagesGenerator,
-	},
-	common: {
-		label: "Example router & proxy Worker",
-		handler: runWorkersGenerator,
-	},
-	scheduled: {
-		label: "Scheduled Worker (Cron Trigger)",
-		handler: (args) => runWorkersGenerator({ ...args, deploy: false }),
-	},
-	queues: {
-		label: "Queue consumer & producer Worker",
-		handler: (args) => runWorkersGenerator({ ...args, deploy: false }),
-	},
-	chatgptPlugin: {
-		label: `ChatGPT plugin`,
-		handler: (args) =>
-			runWorkersGenerator({
-				...args,
-				ts: true,
-			}),
-	},
-	openapi: {
-		label: `OpenAPI 3.1`,
-		handler: (args) =>
-			runWorkersGenerator({
-				...args,
-				ts: true,
-			}),
-	},
-	"pre-existing": {
-		label: "Pre-existing Worker (from Dashboard)",
-		handler: runWorkersGenerator,
-		hidden: true,
-	},
+// A template can have a number of variants, usually js/ts
+type VariantInfo = {
+	path: string;
+};
+
+type StaticFileMap = Record<string, VariantInfo>;
+
+export type FrameworkMap = Awaited<ReturnType<typeof getFrameworkMap>>;
+export type FrameworkName = keyof FrameworkMap;
+
+export const getFrameworkMap = async () => ({
+	angular: (await import("../templates/angular/c3")).default,
+	astro: (await import("../templates/astro/c3")).default,
+	docusaurus: (await import("../templates/docusaurus/c3")).default,
+	gatsby: (await import("../templates/gatsby/c3")).default,
+	hono: (await import("../templates/hono/c3")).default,
+	next: (await import("../templates/next/c3")).default,
+	nuxt: (await import("../templates/nuxt/c3")).default,
+	qwik: (await import("../templates/qwik/c3")).default,
+	react: (await import("../templates/react/c3")).default,
+	remix: (await import("../templates/remix/c3")).default,
+	solid: (await import("../templates/solid/c3")).default,
+	svelte: (await import("../templates/svelte/c3")).default,
+	vue: (await import("../templates/vue/c3")).default,
+});
+
+export const getTemplateMap = async () => {
+	return {
+		"hello-world": await import("../templates/hello-world/c3.json"),
+		// Dummy record
+		webFramework: { displayName: "Website or web app" },
+		common: await import("../templates/common/c3.json"),
+		scheduled: await import("../templates/scheduled/c3.json"),
+		queues: await import("../templates/queues/c3.json"),
+		chatgptPlugin: await import("../templates/chatgptPlugin/c3.json"),
+		openapi: await import("../templates/openapi/c3.json"),
+		"remote-template": { displayName: "From a remote git repository" },
+		"pre-existing": (await import("../templates/pre-existing/c3")).default,
+	} as unknown as Record<string, TemplateConfig>;
```

Comment:
I'm gonna go with option 1.

I see your point about correctness, but I feel like the result of strategy 2 doesn't give us a more correct type. If you evaluate the code paths, `selectTemplate` will always return a `TemplateConfig`, or it will crash. But, by changing the return type of `getTemplateMap` to `TemplateConfig | DummyRecord`, the return type of `selectTemplate` gets inferred to be the same. 

From that point, we either cast the result to `TemplateMap` (what we're trying to avoid for correctness), or defensively handle scenarios in which `selectTemplate `returns a `DummyRecord` (which can't actually happen).